### ISSUE TITLE
DeltaAggregatedResources gRPC config stream to xds_cluster closed: 13,

### ISSUE THREAD

**Issue by Uburro** [envoyproxy/gateway#3559]
*Description*:
Hi all. 
Every 20-30 minutes my pods of envoy recreated. I saw logs

```
[2024-06-06 14:22:33.055][1][warning][main] [source/server/[server.cc:881](https://server.cc:881/)] caught ENVOY_SIGTERM
[2024-06-06 14:22:33.059][1][warning][config] [./source/extensions/config_subscription/grpc/grpc_stream.h:155] DeltaAggregatedResources gRPC config stream to xds_cluster closed: 13,

```

in istio we have keepaliveMaxServerConnectionAge parametrs. I do not find same in envoy gateway. Can you help me please?

**Comment by arkodg** [comment]
looks like you are getting `caught ENVOY_SIGTERM`, are the liveness etc probes failing ?

**Comment by Uburro** [comment]
> looks like you are getting `caught ENVOY_SIGTERM`, are the liveness etc probes failing ?

No, in events i saw that envoy gateway operator make rollout it 

```
5m35s       Normal    EnsuringLoadBalancer     service/envoy-public-efa1f375                  Ensuring load balancer
5m38s       Normal    Scheduled                pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Successfully assigned envoy-gateway-system/envoy-public-efa1f375-76fcc4b64f-l6rtw to infra-eu-north1-a-upyc
5m38s       Normal    SuccessfulCreate         replicaset/envoy-public-efa1f375-76fcc4b64f    Created pod: envoy-public-efa1f375-76fcc4b64f-l6rtw
5m38s       Normal    ScalingReplicaSet        deployment/envoy-public-efa1f375               Scaled up replica set envoy-public-efa1f375-76fcc4b64f to 1 from 0
5m37s       Normal    Pulled                   pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Container image "cr.nemax.nebius.cloud/mirror/io/docker/envoyproxy/envoy:distroless-v1.29.2" already present on machine
5m37s       Normal    Created                  pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Created container envoy
5m37s       Normal    Started                  pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Started container envoy
5m37s       Normal    Pulled                   pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Container image "envoyproxy/gateway-dev:62ff3e7" already present on machine
5m37s       Normal    Created                  pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Created container shutdown-manager
5m37s       Normal    Started                  pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Started container shutdown-manager
5m36s       Normal    ScalingReplicaSet        deployment/envoy-public-efa1f375               Scaled down replica set envoy-public-efa1f375-76fcc4b64f to 0 from 1
5m36s       Normal    Created                  pod/envoy-public-efa1f375-7b499b6b85-xf642     Created container envoy
5m36s       Normal    Scheduled                pod/envoy-public-efa1f375-7b499b6b85-xf642     Successfully assigned envoy-gateway-system/envoy-public-efa1f375-7b499b6b85-xf642 to infra-eu-north1-c-afiw
5m36s       Normal    SuccessfulDelete         replicaset/envoy-public-efa1f375-76fcc4b64f    Deleted pod: envoy-public-efa1f375-76fcc4b64f-l6rtw
5m36s       Normal    ScalingReplicaSet        deployment/envoy-public-efa1f375               Scaled up replica set envoy-public-efa1f375-7b499b6b85 to 1 from 0
5m36s       Normal    Killing                  pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Stopping container shutdown-manager
5m36s       Warning   FailedToUpdateEndpoint   endpoints/envoy-public-efa1f375                Failed to update endpoint envoy-gateway-system/envoy-public-efa1f375: Operation cannot be fulfilled on endpoints "envoy-public-efa1f375": the object has been modified; please apply your changes to the latest version and try again
5m33s       Normal    EnsuredLoadBalancer      service/envoy-public-efa1f375                  Ensured load balancer
5m36s       Normal    ScalingReplicaSet        deployment/envoy-public-efa1f375               Scaled down replica set envoy-public-efa1f375-7b499b6b85 to 0 from 1
5m36s       Normal    SuccessfulDelete         replicaset/envoy-public-efa1f375-7b499b6b85    Deleted pod: envoy-public-efa1f375-7b499b6b85-xf642
5m36s       Normal    SuccessfulCreate         replicaset/envoy-public-efa1f375-7b499b6b85    Created pod: envoy-public-efa1f375-7b499b6b85-xf642
5m36s       Normal    Pulled                   pod/envoy-public-efa1f375-7b499b6b85-xf642     Container image "cr.nemax.nebius.cloud/mirror/io/docker/envoyproxy/envoy:distroless-v1.29.2" already present on machine
5m36s       Normal    Killing                  pod/envoy-public-efa1f375-76fcc4b64f-l6rtw     Stopping container envoy
5m35s       Normal    Pulled                   pod/envoy-public-efa1f375-7b499b6b85-xf642     Container image "envoyproxy/gateway-dev:62ff3e7" already present on machine
5m35s       Normal    Started                  pod/envoy-public-efa1f375-7b499b6b85-xf642     Started container envoy
5m35s       Normal    Killing                  pod/envoy-public-efa1f375-7b499b6b85-xf642     Stopping container shutdown-manager
5m35s       Normal    Started                  pod/envoy-public-efa1f375-7b499b6b85-xf642     Started container shutdown-manager
5m35s       Normal    Created                  pod/envoy-public-efa1f375-7b499b6b85-xf642     Created container shutdown-manager
5m35s       Normal    Killing                  pod/envoy-public-efa1f375-7b499b6b85-xf642     Stopping container envoy
2m7s        Normal    UpdatedLoadBalancer      service/envoy-public-efa1f375                  Updated load balancer with new hosts
```


```
envoy-public-efa1f375-549c4f4775    2         2         2       7d16h
envoy-public-efa1f375-557c79cd68    0         0         0       9h
envoy-public-efa1f375-76d66b5f58    0         0         0       2d9h
envoy-public-efa1f375-76fcc4b64f    0         0         0       14h
envoy-public-efa1f375-77f6977fc9    0         0         0       7d12h
envoy-public-efa1f375-7945d9b799    0         0         0       16h
envoy-public-efa1f375-7b499b6b85    0         0         0       7d23h
envoy-public-efa1f375-84656cc87b    0         0         0       7d23h
envoy-public-efa1f375-8478d6956c    0         0         0       33h
envoy-public-efa1f375-85b46d786c    0         0         0       7d12h
envoy-public-efa1f375-cb7f6954f     0         0         0       28h
```

but i did not find reason in logs envoy operator

**Comment by Uburro** [comment]
> looks like you are getting `caught ENVOY_SIGTERM`, are the liveness etc probes failing ?

it is strange situation

i have main gateway with configuration

```
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: public
spec:
  gatewayClassName: public
  listeners:
  - allowedRoutes:
      namespaces:
        from: All
    name: http
    port: 80
    protocol: HTTP
  - allowedRoutes:
      namespaces:
        from: All
    name: https
    port: 443
    protocol: TLS
    tls:
      mode: Passthroughh
```

and a lot tof gateway like this

```
spec:
  gatewayClassName: public
  listeners:
  - allowedRoutes:
      namespaces:
        from: All
    hostname: tracking.mlflow-test.cloud
    name:racking.mlflow-test.cloud
    port: 443
    protocol: HTTPS
    tls:
      certificateRefs:
      - group: ""
        kind: Secret
        name: cert-tracking-secret
      mode: Terminate
```


and saw how periodical;y changing rolling upgrade and changed ports

```
        - containerPort: 10443
          name: mlflow-cf75fb44
          protocol: TCP
```
```
        - containerPort: 10443
          name: envoy-e3314cb2
          protocol: TCP
```

kind: EnvoyProxy have option mergeGateways: true

**Comment by arkodg** [comment]
hey @Uburro you're hitting https://github.com/envoyproxy/gateway/pull/3130, this should be part of the next patch release out soon

**Comment by Uburro** [comment]
> hey @Uburro you're hitting #3130, this should be part of the next patch release out soon

Thank you!

**Comment by shawnh2** [comment]
Should be fixed by #3130, close this one now. Reopen if got any issues.
